circuit.py

```diff
@@ -116,12 +116,24 @@ def commute_rotation(self, index: int) -> None:
 
         next_block = index + 1
         if not self.are_commuting(index, next_block):
+            # for tracking # of i (see issue #28)
+            # i is required for an anti-commute operators pair
+            i_count = 0     
+            
             for i in range(self.qubit_num):
-                new_op = PauliOperator.multiply_by_i(self.ops[index].get_op(i), self.ops[next_block].get_op(i))
-                self.ops[next_block].change_single_op(i, new_op)
+                new_op = PauliOperator.are_commuting(self.ops[index].get_op(i), self.ops[next_block].get_op(i))
+                
+                self.ops[next_block].change_single_op(i, new_op[1])
+                if not new_op[0]: i_count += 1
             
+            # Flip the phase if the current rotation's phase is negative 
             if self.ops[index].rotation_amount < 0:
```

Comment:
Sorry I can't remember why we have to do this flip

Reply:
I suspect this was a temporary hack of some kind

Reply:
![00336F56-A85A-4B27-98D3-1FB741B3EA35](https://user-images.githubusercontent.com/36427091/109399510-83722e00-7943-11eb-89bc-0a31b1da7d73.png)
When the operations commute there is no problem with the negative angle, I guess all we have to do is check the non commuting case

Reply:
We don't have to worry about the commute case since we are not required to calculate iPP' (as it is just a straight swap). The line is run only when P and P' are anti-commute.

Reply:
Further discussion in #30.